include/mm/slab.h

```diff
@@ -74,11 +76,40 @@ struct meta_slab {
     list_head_t slab_head;
     void *slab_base;
     unsigned int slab_len;
-    unsigned int slab_size;
+/*
+* Don't need more than 12 bits. Currently max slab size is 2048 bytes = 2^11
+*/
+    unsigned int slab_size : 12;
+/*
+* slab_allocs is tracking number of allocations currently in this slab. 
+* At max this can go 4096/16 = 256 slabs. Thus 10 bits are enough
+*/    
+    unsigned int slab_allocs : 10;
+    unsigned int reserved : ((sizeof(unsigned int)*8) - 22);
 };
 
 typedef struct meta_slab meta_slab_t;
 
+static inline void increment_slab_allocs(meta_slab_t *slab) {
+    ASSERT(slab != NULL);
+    ASSERT((slab->slab_allocs < MAX_SLAB_ALLOC_COUNT));
+    ASSERT((slab->slab_allocs < (PAGE_SIZE/slab->slab_size)));
```

Comment:
I would turn the division into a macro with page size as parameter.
Also, as ASSERTs sooner or later will not be compiled-in, this could be a BUG_ON